pynonymizer/database/mssql/__init__.py

```diff
@@ -208,7 +208,10 @@ def __get_column_subquery(self, column_strategy):
         elif column_strategy.strategy_type == UpdateColumnStrategyTypes.UNIQUE_LOGIN:
             return f"( SELECT NEWID() )"
         elif column_strategy.strategy_type == UpdateColumnStrategyTypes.FAKE_UPDATE:
-            return f"( SELECT TOP 1 [{column_strategy.qualifier}] FROM [{SEED_TABLE_NAME}] ORDER BY NEWID())"
+            column = column_strategy.qualifier
+            if column_strategy.sql_type:
```

Comment:
i'm pretty sure this syntax won't work for mssql, either. It needs to be some variation on `CAST(...` or `CONVERT`.

Check it out in sqlfiddle if you're not sure, maybe?